stats/stats.go

```diff
@@ -135,13 +135,24 @@ var (
 		2000, 3000, 4000, 5000, 7500, 10000, // another order of magnitude coarsly covered
 		20000, 30000, 40000, 50000, 75000, 100000, // ditto, the end
 	}
-	numValues  = len(histogramBucketValues)
-	numBuckets = numValues + 1 // 1 special first bucket is <= 0; and 1 extra last bucket is > 100000
-	firstValue = float64(histogramBucketValues[0])
-	lastValue  = float64(histogramBucketValues[numValues-1])
-	val2Bucket []int
+	numValues              = len(histogramBucketValues)
+	numBuckets             = numValues + 1 // 1 special first bucket is <= 0; and 1 extra last bucket is > 100000
+	firstValue             = float64(histogramBucketValues[0])
+	lastValue              = float64(histogramBucketValues[numValues-1])
+	line5EndToEndLength    = 13
+	line5EndIndex          = numValues - line5EndToEndLength //to reach 1000
+	constantTimeVal2Bucket []int                             //up to 1000
+	linearTimeVal2Bucket   []UpperValueIndexer               // later than 1000 to 100000
 )
 
+// UpperValueIndexer holds upper values of histogramBucketValues such as 1000 to 100000
+// Index and Scope are kept. Scope is estimated with current index value - previous index value plus previous
+// sum of scopes.
+type UpperValueIndexer struct {
```

Comment:
I don;'t think this is needed - we just need the index for maxArrayValue in 	histogramBucketValues